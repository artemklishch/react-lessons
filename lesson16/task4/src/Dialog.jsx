import React from 'react';
import PropTypes from 'prop-types';

const Dialog = ({ isOpen, children, onClose, title }) => {
  if(!isOpen)return null;
  return (
    <div className="dialog">
      <div className="dialog__heading">
        <h4 className="dialog__title">{title}</h4>
        <button className="dialog__close-btn" onClick={onClose}>+</button>
      </div>
      <div className="dialog__content">{children}</div>
    </div>
  );
};
export default Dialog;

Dialog.propTypes = {
  isOpen: PropTypes.bool,
  children: PropTypes.element.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string,
};
Dialog.defaulrProps = {
  isOpen: false,
  title: ''
};
