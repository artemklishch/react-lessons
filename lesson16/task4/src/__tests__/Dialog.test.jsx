import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../Dialog';

describe('Dialog', () => {
  it ('should not render Dialog in case of isOpen false', () => {
    const child = <p>Some text</p>
    const props = {
      isOpen: false,
      children: child, 
      onClose: jest.fn(), 
      title: 'Title',
    };
    const wrappedComponent = shallow(<Dialog {...props} />);
    expect(wrappedComponent.find('.dialog').exists()).toEqual(false);
  })

  it ('should not render Dialog in case of isOpen false', () => {
    const child = <p>Some text</p>
    const props = {
      isOpen: true,
      children: child, 
      onClose: jest.fn(), 
      title: 'Title',
    };
    const wrappedComponent = shallow(<Dialog {...props} />);
    // expect(wrappedComponent.find('.dialog').exists()).toEqual(true);
    expect(wrappedComponent).toBeTruthy();
  })

  it ('should not render Dialog in case of isOpen false', () => {
    const child = <p>Some text</p>
    const mockOnClose = jest.fn();
    const props = {
      isOpen: true,
      children: child, 
      title: 'Title',
    };
    const wrappedComponent = shallow(<Dialog onClose={mockOnClose} {...props} />);
    wrappedComponent.find('.dialog__close-btn').simulate('click');
    expect(mockOnClose).toBeCalled();
  })

  it ('should render Dialog with its elements', () => {
    const child = <p>Some text</p>;
    const props = {
      isOpen: true,
      children: child, 
      onClose: jest.fn(), 
      title: 'Title',
    };
    const wrappedComponent = shallow(<Dialog {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  })
});