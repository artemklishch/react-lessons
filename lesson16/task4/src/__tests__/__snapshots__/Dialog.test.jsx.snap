// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render Dialog with its elements 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    >
      Title
    </h4>
    <button
      className="dialog__close-btn"
      onClick={[MockFunction]}
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  >
    <p>
      Some text
    </p>
  </div>
</div>
`;
